Add css variable support and use MD icon

skjnldsv opened the pull request:
1. Migrate the icon to material design
2. Switch to css variable compatibility
3. Use `EmptyContent`
4. Use `getCapabilities`
5. use css to animate

Instant change on theming update:
![Peek 31-03-2022 18-27](https://user-images.githubusercontent.com/14975046/161104599-326f9683-7d13-4868-b76a-fa18366bfd21.gif)

![image](https://user-images.githubusercontent.com/14975046/164729106-fd49c8d8-d7ea-426b-98ac-58829521a786.png)


TODO in follow-up:
- [ ] Migrate the scss file to inner vue css file
- [ ] Cleanup old styling and vars usage

nickvergessen commented on src/App.vue:
```diff
@@ -29,21 +42,23 @@
+					{{ webNotificationsGranted === null
+						? t('notifications', 'Requesting browser permissions to show notifications')
+						: t('notifications', 'No notifications') }}
+					<template #icon><Bell /></template>
```
```suggestion
					<template #icon><Bell title="" decorative /></template>
```

nickvergessen commented on package.json:
```diff
@@ -21,11 +21,10 @@
 		"@nextcloud/capabilities": "^1.0.4",
 		"@nextcloud/dialogs": "^3.1.2",
 		"@nextcloud/event-bus": "^2.1.1",
-		"@nextcloud/initial-state": "^1.2.1",
```
still needed in settings

nickvergessen commented:
The icon of notifications are not correctly depending on the background anymore
And the empty content view looks a bit weird

Before | After
---|---
![Bildschirmfoto von 2022-04-01 11-31-41](https://user-images.githubusercontent.com/213943/161237248-095c6d3d-258f-484a-921a-8a6057bc014d.png) | ![Bildschirmfoto von 2022-04-01 11-30-34](https://user-images.githubusercontent.com/213943/161237266-dc7080b8-ecc8-4de6-ad5f-25dff96dd84c.png)
![Bildschirmfoto von 2022-04-01 11-31-58](https://user-images.githubusercontent.com/213943/161237282-87df9973-b21c-48fc-87d1-abdc0295994a.png) | ![Bildschirmfoto von 2022-04-01 11-32-12](https://user-images.githubusercontent.com/213943/161237300-ad7f1552-0547-4776-b9c2-b5e5f7172e3d.png)

skjnldsv commented:
> The icon of notifications are not correctly depending on the background anymore

How does that works?
How is the Notification API handling this?

nickvergessen commented:
https://github.com/nextcloud/notifications/blob/master/css/styles.scss#L148-L150

I guess this does not match anymore

> How is the Notification API handling this?

The API can not handle it as the icon needs to be neutral for the mobile apps and desktop client so they can individually apply the dark mode behaviour.

skjnldsv commented:
> [https://github.com/nextcloud/notifications/blob/master/css/styles.scss#L148-L150](https://github.com/nextcloud/notifications/blob/master/css/styles.scss?rgh-link-date=2022-04-04T20%3A25%3A42Z#L148-L150)
> 
> I guess this does not match anymore

Nice, I guess we can do this straight away in 25 once the scss cacher pr is merged

skjnldsv commented:
Done @nickvergessen

szaimen reviewed (changes requested):
Seems to work but didn't review the code.

Please adjust https://github.com/nextcloud/notifications/blob/6f9f9b50c30820f250d937a92a1128994b324e1d/css/styles.scss#L63 to `max-height: calc(100vh - 50px * 4 - 80px) !important;`

in order to fix 
![image](https://user-images.githubusercontent.com/42591237/164733634-794049f0-7cf7-43c5-a654-5a4f3fa77ece.png)

skjnldsv commented:
Fixed and fixed the animation too
![Peek 22-04-2022 17-46](https://user-images.githubusercontent.com/14975046/164749353-f5f5155c-9147-41f3-a23f-cb7eb327f99e.gif)

skjnldsv commented:
/compile amend /

szaimen reviewed (approved):
Looks good from the screencast
